src/main/kotlin/uk/gov/gdx/datashare/models/validators/SingleConsumptionMethodValidator.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.gdx.datashare.models.validators
+
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import uk.gov.gdx.datashare.models.AcquirerSubRequest
+
+class SingleConsumptionMethodValidator : ConstraintValidator<SingleConsumptionMethod, AcquirerSubRequest> {
+  override fun isValid(value: AcquirerSubRequest, context: ConstraintValidatorContext): Boolean {
+    if (value.oauthClientId == null && value.queueName == null) {
+      return false
+    }
+
+    if (value.oauthClientId != null && value.queueName != null) {
+      return false
+    }
+
+    if ((value.queueName == null && value.principalArn != null) ||
+      (value.queueName != null && value.principalArn == null)
+    ) {
+      context.disableDefaultConstraintViolation()
```

Comment:
This constraint + the naming in the annotation feels a little bit inconsistent, can I double check what the overall aim of it is?

Reply:
I _think_ we're saying you must give us one and only one of
- oauth client ID
- queue name+principal arn
Is that right?

Reply:
Yes. I'm splitting this out into two constraints and validators - it's really two concepts. If you're happy for it to come in a followup with the /admin stuff I'll do it there, if not I'll change this